pep_sphinx_extensions/pep_zero_generator/pep_0_parser.py

```diff
@@ -0,0 +1,288 @@
+        pep_type : The type of PEP.  Can only be one of the values from PEP.type_values.
+        status : The PEP's status.  Value must be found in PEP.status_values.
+        authors : A list of the authors.
+
+    """
+
+    # The required RFC 822 headers for all PEPs.
+    required_headers = {"PEP", "Title", "Author", "Status", "Type", "Created"}
+
+    # Valid values for the Type header.
+    type_values = {"Standards Track", "Informational", "Process"}
+    # Valid values for the Status header.
+    # Active PEPs can only be for Informational or Process PEPs.
+    status_values = {
+        "Accepted", "Provisional", "Rejected", "Withdrawn",
+        "Deferred", "Final", "Active", "Draft", "Superseded",
+    }
+
+    def raise_pep_error(self, msg: str, pep_num: bool = False) -> None:
+        pep_number = self.number if pep_num else None
+        raise PEPError(msg, self.filename, pep_number=pep_number)
+
+    def __init__(self, filename: Path, author_lookup: dict, title_length: int):
+        """Init object from an open PEP file object.
+
+        pep_file is full text of the PEP file, filename is path of the PEP file, author_lookup is author exceptions file
+
+        """
+        self.filename: Path = filename
+        self.title_length: int = title_length
+
+        # Parse the headers.
+        pep_text = filename.read_text("UTF8")
+        metadata = HeaderParser().parsestr(pep_text)
+        required_header_misses = self.required_headers - set(metadata.keys())
+        if required_header_misses:
+            msg = f"PEP is missing required headers ({', '.join(required_header_misses)})"
+            self.raise_pep_error(msg)
+
+        try:
+            self.number: int = int(metadata["PEP"])
+        except ValueError:
+            self.raise_pep_error("PEP number isn't an integer")
+
+        # Check PEP number matches filename
+        if self.number != int(filename.stem[4:]):
+            self.raise_pep_error(f"PEP number does not match file name ({filename})", pep_num=True)
+
+        # Title
+        self.title: str = metadata["Title"]
+
+        # Type
+        self.pep_type: str = metadata["Type"]
+        if self.pep_type not in self.type_values:
+            self.raise_pep_error(f"{self.pep_type} is not a valid Type value", pep_num=True)
+
+        # Status
+        status = metadata["Status"]
+        if status not in self.status_values:
+            if status == "April Fool!":  # See PEP 401 :)
+                status = "Rejected"
+            else:
+                self.raise_pep_error(f"{status} is not a valid Status value", pep_num=True)
+
+        # Special case for Active PEPs.
+        if status == "Active" and self.pep_type not in {"Process", "Informational"}:
+            msg = "Only Process and Informational PEPs may have an Active status"
+            self.raise_pep_error(msg, pep_num=True)
+
+        # Special case for Provisional PEPs.
+        if status == "Provisional" and self.pep_type != "Standards Track":
```

Comment:
I think we should capture all these strings in some global vars in case we want to change them in all the document at once. Is a bit of a nit, but I am normally a bit wary of "magical constants" :)

Reply:
done